src/client/camera.cpp

```diff
@@ -137,13 +138,13 @@ bool Camera::successfullyCreated(std::string &error_message)
 	if (!m_playernode) {
 		error_message = "Failed to create the player scene node";
 	} else if (!m_headnode) {
-		error_message = "Failed to create the head scene node";
+		error_message = gettext("Failed to create the head scene node");
```

Comment:
you still have more of these

Reply:
You can delete `successfullyCreated` - these things will never be null, so this is pointless

Reply:
I've reviewed codes. I can confirm the m_playernode, m_headnode, m_cameranode, m_wieldmgr and m_wieldnode have always exists.  So the error_message is never useful indeed.

But I can not remove the `successfullyCreated` method, for need it to trigger the `on_camera_ready` event.

Reply:
Well, I have seen you move the event trigger to the game.cpp.